Sources/TensorFlow/Core/Tensor.swift

```diff
@@ -226,9 +228,14 @@ public extension Tensor {
     /// - Parameters:
     ///   - shape: The shape of the tensor.
     ///   - scalars: The scalar contents of the tensor.
-    /// - Precondition: The number of scalars must equal the product of the dimensions of the shape.
+    /// - Precondition: The product of the dimensions of the shape must equal the number of scalars.
     @inlinable
     init(shape: TensorShape, scalars: [Scalar]) {
+        precondition(shape.contiguousSize == scalars.count,
+            """
+            The product of the dimensions of the shape must equal the number of scalars: \
+            \(shape.contiguousSize) vs. \(scalars.count)
```

Comment:
This is looking really great. One last thought: what do you think of an error message like the following?

```suggestion
            the shape requires \(shape.contiguousSize) scalars but only \(scalars.count) were provided.
```

Reply:
I do like this message better. The mapping from arguments to their problematic values is clearer. I'll update them!